File: OPENAPI-REST-API/openapi-client/javascript-closure-angular/API/Client/ServerOrderMemory.js
goog.provide('API.Client.ServerOrderMemory');

/**
 * A memory (RAM) option available when ordering a dedicated server.
 * @record
 */
API.Client.ServerOrderMemory = function() {}

/**
 * Memory ID.
 * @type {!string}
 * @export
 */
API.Client.ServerOrderMemory.prototype.id;

/**
 * Memory price.
 * @type {!string}
 * @export
 */
API.Client.ServerOrderMemory.prototype.price;

/**
 * Memory image.
 * @type {!string}
 * @export
 */
API.Client.ServerOrderMemory.prototype.img;

/**
 * Short description of the memory.
 * @type {!string}
 * @export
 */
API.Client.ServerOrderMemory.prototype.shortDesc;

/**
 * Long description of the memory.
 * @type {!string}
 * @export
 */
API.Client.ServerOrderMemory.prototype.longDesc;

/**
 * Manufacturer information.
 * @type {!string}
 * @export
 */
API.Client.ServerOrderMemory.prototype.manu;

/**
 * Memory size.
 * @type {!string}
 * @export
 */
API.Client.ServerOrderMemory.prototype.size;

/**
 * Memory type.
 * @type {!string}
 * @export
 */
API.Client.ServerOrderMemory.prototype.type;

/**
 * Hidden status.
 * @type {!string}
 * @export
 */
API.Client.ServerOrderMemory.prototype.hidden;

/**
 * Monthly price.
 * @type {!number}
 * @export
 */
API.Client.ServerOrderMemory.prototype.monthlyPrice;

/**
 * Drive type.
 * @type {!string}
 * @export
 */
API.Client.ServerOrderMemory.prototype.driveType;

/**
 * Display of monthly memory price.
 * @type {!string}
 * @export
 */
API.Client.ServerOrderMemory.prototype.monthlyPriceDisplay;

File: OPENAPI-REST-API/swagger-client/scala-akka-http-server/src/main/scala/io/swagger/server/model/ServerOrderMemory.scala
package io.swagger.server.model


/**
 * A memory (RAM) option available when ordering a dedicated server.
 *
 * @param id Memory ID. for example: ''65''
 * @param price Memory price. for example: ''0''
 * @param img Memory image. for example: ''hd.jpg''
 * @param short_desc Short description of the memory. for example: ''20TB SATA''
 * @param long_desc Long description of the memory.
 * @param manu Manufacturer information.
 * @param size Memory size. for example: ''3000''
 * @param `type` Memory type.
 * @param hidden Hidden status. for example: ''0''
 * @param monthly_price Monthly price. for example: ''50''
 * @param drive_type Drive type. for example: ''lff''
 * @param monthly_price_display Display of monthly memory price. for example: ''$50.00''
 */
case class ServerOrderMemory (
  id: Option[String],
  price: Option[String],
  img: Option[String],
  short_desc: Option[String],
  long_desc: Option[String],
  manu: Option[String],
  size: Option[String],
  `type`: Option[String],
  hidden: Option[String],
  monthly_price: Option[Int],
  drive_type: Option[String],
  monthly_price_display: Option[String]
)

